semantics/language/execution/compound-literal.k

```diff
@@ -57,10 +57,23 @@ module C-COMPOUND-LITERAL
                ~> X
           ...</k>
           <compound-lit-map> M:Map => M[X <- Base] </compound-lit-map>
+          <local-addresses>... .Set => SetItem(Base) </local-addresses>
```

Comment:
This adds the newly allocated address to the set of local addresses so it'll be cleaned up/invalidated when the compound literal object goes out of scope. Compound literals at block scope are supposed to have a lifetime of the block.